packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts

```diff
@@ -1,26 +1,29 @@
 import { TemplateModuleInternal } from "../../../../common/src/template/internal/types.js";
-import {
-  renderHeadConfigToString,
-  getLang,
-} from "../../../../common/src/template/head.js";
+import { getLang } from "../../../../common/src/template/head.js";
 import {
   TemplateRenderProps,
-  GetHeadConfig,
+  RenderTemplate,
 } from "../../../../common/src/template/types.js";
+import {
+  getHydrationTemplate,
+  getServerTemplatePlugin,
+} from "../../../../common/src/template/hydration.js";
+import path from "path";
 
-export const reactWrapper = <T extends TemplateRenderProps>(
+export const reactWrapper = async <T extends TemplateRenderProps>(
   props: T,
   templateModuleInternal: TemplateModuleInternal<any, any>,
-  template: string,
   hydrate: boolean,
-  getHeadConfig?: GetHeadConfig<any>
-): string => {
+  clientRenderPath: string,
```

Comment:
the server module should be imported ASAP and cached similar to what we do with the TemplateModules. The main reason for doing so is that dynamically importing without caching causes a memory leak in the plugin. A secondary reason is that it'd be clearer.

Reply:
You mean load it in renderer.ts instead? Good point. We don't need to keep loading these over and over. I will update. Before I do that I want to confirm some of the other questions first, like having a RenderTemplateInternal type.